### PR TITLE
Add mobile header banner

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR inserts a `<header>` (banner) element to the mobile UI and moves the `<main>` region so that the `<header>` is outside of it (so that "Skip navigation" actually skips the header content). In some screens, the map component is moved after the more important content to facilitate keyboard and assistive technologies.

Out of scope: applicable `<h1>` headings missing from some of the screens.

Some files are converted to TypeScript and some components are converted to function components.

### CODE REVIEW

**Review by amy-corson-ibigroup — APPROVED** [review]
Quick clarifying question, but otherwise this look great!

File: lib/components/mobile/route-viewer.js [hunk 1]
```diff
@@ -32,27 +32,27 @@ class MobileRouteViewer extends Component {
...
+        <main tabIndex={-1}>
+          <div className="viewer-container">
+            <RouteViewer hideBackButton ModeIcon={ModeIcon} />
+          </div>
+
+          {/* The map is less important semantically, so keyboard focus and screen readers
+              will focus on the route viewer first. The map will still appear first visually. */}
+          <div className="viewer-map">
+            <DefaultMap />
+          </div>
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/mobile/route-viewer.js:44]: Would love to, however each viewer sets the title bar. We would have to touch more code to do that, so I think that can be in a separate PR.

[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/mobile/route-viewer.js:44]: Makes sense, thanks!

**Review by miles-grant-ibigroup — APPROVED** [review]
Thanks for all this cleanup work! Works great for me and winds up being much cleaner

**Pull request merged**